### PR TITLE
Add support for multi touches

### PR DESCRIPTION
**Pull request by dr1rrb**
GitHub Issue:
fixes https://github.com/unoplatform/uno/issues/2092
fixes https://github.com/unoplatform/uno/issues/2219

## Bugfix
- Fix horizontal scrolling in `ListView`
- Pointer capture

## Feature
Add support of multi-touch

## What is the current behavior?
- Multi touch was not supported
- The fix https://github.com/unoplatform/uno/pull/2338 broke the manipulations event while in `ScrollViewer` ...
- ... and underlined a bug were capture (and manipulations) was released prematurely

## What is the new behavior?
- Synchronous multiple pointers are now properly differentiated, so multi-touch manipulations are now detected properly
- The captures are now longer released too early, so the horizontal scroll in SV work properly

## PR Checklist
- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

### CODE REVIEW

**Comment by dr1rrb** [comment]
This is draft until comments of https://github.com/unoplatform/uno/pull/2392 has been integrated in this PR

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Input/Manipulation_Tests.cs [hunk 1]
```diff
@@ -23,5 +26,72 @@ public void TestManipulation()
 
 			TakeScreenshot("Result");
 		}
+
+		[Test]
+		[AutoRetry]
+		[ActivePlatforms(Platform.Android, Platform.iOS)] // PinchToZoomInCoordinates is not supported on WASM yet
```
Comment (jeromelaban) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Input/Manipulation_Tests.cs]: Open an issue for that one

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Input/Manipulation_Tests.cs [hunk 2]
```diff
@@ -23,5 +26,72 @@ public void TestManipulation()
...
+				Rotation = f("Δr"),
+				Scale = f("Δs"),
+				Expansion = f("Δe")
+			};
+			var cumulative = new ManipulationDelta
+			{
+				Translation = p("ΣtrX", "ΣtrX"),
+				Rotation = f("Σr"),
+				Scale = f("Σs"),
+				Expansion = f("Σe")
```
Comment (carldebilly) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Input/Manipulation_Tests.cs]: 𝓟𝓵𝓪𝔂𝓲𝓷𝓰 with 𝕦𝕟𝕚𝕔𝕠𝕕𝕖 ;-)

File: src/Uno.UI/UI/Xaml/Input/PointerRoutedEventArgs.Android.cs [hunk 3]
```diff
@@ -15,17 +15,29 @@ namespace Windows.UI.Xaml.Input
 {
 	partial class PointerRoutedEventArgs
 	{
+		private const int _pointerIdsCount = (int)MotionEventActions.PointerIndexMask >> (int)MotionEventActions.PointerIndexShift; // 0xff
+		private const int _pointerIdsShift = 31 - (int)MotionEventActions.PointerIndexShift; // 23
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Input/PointerRoutedEventArgs.Android.cs]: You should document those magic numbers

File: src/Uno.UI/UI/Xaml/Input/PointerRoutedEventArgs.Android.cs [hunk 4]
```diff
@@ -39,39 +51,52 @@ public PointerPoint GetCurrentPoint(UIElement relativeTo)
...
 
-				return posRelToTarget;
+				relative = posRelToTarget;
 			}
+
+			// Raw coordinates are relative to the screen (easier for the gesture recognizer to track fingers for manipulations)
+			// if (ANDROID > 10)
+			// {
+			//		var raw = new Point(_nativeEvent.getRawX(_pointerIndex), _nativeEvent.getRawY(_pointerIndex)).PhysicalToLogicalPixels();
+			// }
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Input/PointerRoutedEventArgs.Android.cs]: Could be `__ANDROID_10__`

File: src/Uno.UI/UI/Xaml/Input/PointerRoutedEventArgs.Android.cs (hunk 3 above)
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Input/PointerRoutedEventArgs.Android.cs]: It's documented where I use those values (cf. line 31)

File: src/Uno.UI/UI/Xaml/Input/PointerRoutedEventArgs.Android.cs (hunk 4 above)
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Input/PointerRoutedEventArgs.Android.cs]: True ... it's only pseudo code for now as we don't have Android 10 support yet

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Input/Manipulation_Tests.cs (hunk 1 above)
Comment (dr1rrb) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Input/Manipulation_Tests.cs]: https://github.com/unoplatform/Uno.UITest/issues/27

File: doc/ReleaseNotes/_ReleaseNotes.md [hunk 5]
```diff
@@ -11,6 +11,7 @@
 - Add clarification in [documentation](../articles/uno-development/working-with-the-samples-apps.md) for adding automated UI tests
 - Add support for `Popup.LightDismissOverlayMode`, as well as `DatePicker.LightDismissOverlayMode` and `Flyout.LightDismissOverlayMode`
 - `TransformToVisual` now returns a real transform to convert coordinates between views (was only returning a translate transform to offset the origin of controls)
+- Multiple pointers at same time on screen (a.k.a. Multi-touch) are now supported
```
[the remark this answers is not in the document]
Comment (dr1rrb) [reply doc/ReleaseNotes/_ReleaseNotes.md]: I didn't explicitly tested this case, but yes: the main issue was that previously the pointer IDs was wrongly calculated and fingers A and B were sometimes confused. So usually they was considered just as if a single pointer which was moving incredibly fast on the screen between 2 locations, so in your example neither button A or B would have raised the`click` event.

Now as they are really considered as 2 independent pointers, in you example, you should get `B.clicked` then (if you didn't moved your finger out of button A while holding) `A.clicked`.

And for same reason, the `ManipulationDelta.Scale` is now properly computed, so yes you are able to "pinch" something.

**Pull request merged**